src/cli/cli_network_data.cpp

```diff
@@ -244,72 +244,100 @@ template <> otError NetworkData::Process<Cmd("publish")>(Arg aArgs[])
              * @code
-             * netdata publish dnssrp unicast 50152
+             * netdata publish dnssrp unicast 50152 1
              * Done
              * @endcode
-             * @cparam netdata publish dnssrp unicast @ca{port}
+             * @cparam netdata publish dnssrp unicast @ca{port} [@ca{version}]
              * @par
-             * Publishes the device's Mesh-Local EID with a port number. MLE and port information is
-             * included in the Server TLV data. To use a different Unicast address, use the
-             * `netdata publish dnssrp unicast (addr,port)` command.
+             * Publishes the device's Mesh-Local EID with a port number and given version. MLE, port and version
+             * information is included in the Server TLV data. To use a different Unicast address, use the
+             * `netdata publish dnssrp unicast (addr,port,version)` command.
              * @par
              * Any current DNS/SRP Service entry being published from a previous
              * `publish dnssrp{anycast|unicast}` command is removed and replaced with the new arguments.
              * @par
              * `OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE` must be enabled.
-             * @csa{netdata publish dnssrp unicast (addr,port)}
+             * @csa{netdata publish dnssrp unicast (addr,port,version)}
              * @csa{netdata publish dnssrp anycast}
              * @sa otNetDataPublishDnsSrpServiceUnicastMeshLocalEid
              */
-            if (aArgs[3].IsEmpty())
+            if (!hasAddress)
             {
-                SuccessOrExit(error = aArgs[2].ParseAsUint16(port));
-                otNetDataPublishDnsSrpServiceUnicastMeshLocalEid(GetInstancePtr(), port);
+                otNetDataPublishDnsSrpServiceUnicastMeshLocalEid(GetInstancePtr(), port, version);
                 ExitNow();
             }
 
             /**
-             * @cli netdata publish dnssrp unicast (addr,port)
+             * @cli netdata publish dnssrp unicast (addr,port,version)
              * @code
-             * netdata publish dnssrp unicast fd00::1234 51525
+             * netdata publish dnssrp unicast fd00::1234 51525 1
              * Done
              * @endcode
-             * @cparam netdata publish dnssrp unicast @ca{address} @ca{port}
+             * @cparam netdata publish dnssrp unicast @ca{address} @ca{port} [@ca{version}]
              * @par
-             * Publishes a DNS/SRP Service Unicast Address with an address and port number. The address
-             * and port information is included in Service TLV data. Any current DNS/SRP Service entry being
+             * Publishes a DNS/SRP Service Unicast Address with an address and port and version number. The address,
+             * port, and version information is included in Service TLV data. Any current DNS/SRP Service entry being
              * published from a previous `publish dnssrp{anycast|unicast}` command is removed and replaced
```

Comment:
```suggestion
             * published from a previous `publish dnssrp {anycast|unicast}` command is removed and replaced
```

Reply:
Fixed these and also rebased on new push. Thanks.